sdk/client/src/verifying_client.rs

```diff
@@ -421,69 +439,139 @@ fn verifying_get_account(address: AccountAddress, version: Option<Version>) -> V
-                .map_err(Error::decode)?;
-            let currency_codes = currencies
-                .as_ref()
-                .map(RegisteredCurrencies::currency_codes)
-                .ok_or_else(|| Error::unknown("diem_root has no registered currencies"))?;
-
-            let maybe_account_view = account_state_with_proof
-                .blob
-                .map(|blob| {
-                    let account_state = AccountState::try_from(&blob)
-                        .map_err(Error::decode)?;
-                    AccountView::try_from_account_state(
-                        address,
-                        account_state,
-                        &currency_codes,
-                        version,
-                    ).map_err(Error::decode)
-                })
-                .transpose()?;
-
-            Ok(MethodResponse::GetAccount(maybe_account_view))
-        }
-        subresponses => return Err(Error::rpc_response(format!(
+        let (diem_root, account) = match subresponses {
+            [MethodResponse::GetAccountStateWithProof(ref diem_root), MethodResponse::GetAccountStateWithProof(ref account)] =>
+                (diem_root, account),
+            subresponses => {
+                return Err(Error::rpc_response(format!(
                     "expected [GetAccountStateWithProof, GetAccountStateWithProof] subresponses, received: {:?}",
                     subresponses,
-                    ))),
-    }
+                )))
+            }
+        };
+
+        let diem_root_with_proof =
+            AccountStateWithProof::try_from(diem_root).map_err(Error::decode)?;
+        let account_state_with_proof =
+            AccountStateWithProof::try_from(account).map_err(Error::decode)?;
+
+        let (address, version) = match ctxt.request {
+            MethodRequest::GetAccount(address, version) => (*address, *version),
+            request => panic!("programmer error: unexpected request: {:?}", request),
+        };
+        let latest_li = ctxt.state_proof.0.ledger_info();
+        let ledger_version = latest_li.version();
+        let version = version.unwrap_or(ledger_version);
+
+        diem_root_with_proof
+            .verify(latest_li, version, diem_root_address())
+            .map_err(Error::invalid_proof)?;
+        account_state_with_proof
+            .verify(latest_li, version, address)
+            .map_err(Error::invalid_proof)?;
+
+        // TODO(philiphayes): it seems wasteful to lookup the whole diem_root
+        // account, verify the proofs, etc... just to get a list of supported
+        // currency codes. Would it make sense for the AccountView to just
+        // list _all_ BalanceResource's under balance? Then we could avoid
+        // this extra lookup. It's even more problematic when a batch
+        // includes many GetAccount requests, as this batch handling logic
+        // isn't (currently) smart enough to dedup the diem_root lookups.
```

Comment:
wondering what is this todo for

Reply:
ahh, if you look at that commit, it's just a reformatting commit (changing the indentation).

as for the comment, it's about more efficiently extracting the balance resources from an account without having to first look up the global supported currencies in DiemRoot.

Reply:
So, sometime, you do leave a TODO :D

Reply:
#8415 takes care of this TODO :^)